rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/NestedParticipantIT.java

```diff
@@ -99,34 +106,71 @@ public void nestedParticipantAfterLRACallTest() {
                 .header(LRA.LRA_HTTP_CONTEXT_HEADER, parentLRA)
                 .get();
 
-            Assert.assertEquals(200, response.getStatus());
+            assertEquals(200, response.getStatus());
             Assert.assertTrue(response.hasEntity());
 
             nestedLRA = URI.create(response.readEntity(String.class));
-            Assert.assertNotEquals(parentLRA, nestedLRA);
-            Assert.assertEquals(1, lraMetricService.getMetric(LRAMetricType.Nested, parentLRA, NestedParticipant.class));
+            assertNotEquals(parentLRA, nestedLRA);
+            assertEquals(1, lraMetricService.getMetric(LRAMetricType.Nested, parentLRA, NestedParticipant.class));
         } finally {
             if (response != null) {
                 response.close();
             }
         }
 
         // close nested LRA
-        narayanaLRAClient.closeLRA(nestedLRA);
+        lraClient.closeLRA(nestedLRA);
         // the nested LRA should be in Closed state, however, we keep it in Closing state
         // so we can't wait for the recovery of the nested LRA
         // https://issues.redhat.com/browse/JBTM-3330
         narayanaLRARecovery.waitForEndPhaseReplay(nestedLRA);
 
-        Assert.assertEquals(1, lraMetricService.getMetric(LRAMetricType.Completed, nestedLRA, NestedParticipant.class));
-        Assert.assertEquals(2, lraMetricService.getMetric(LRAMetricType.Nested, parentLRA, NestedParticipant.class));
-        Assert.assertEquals(0, lraMetricService.getMetric(LRAMetricType.AfterLRA, nestedLRA, NestedParticipant.class));
+        assertEquals(1, lraMetricService.getMetric(LRAMetricType.Completed, nestedLRA, NestedParticipant.class));
+        assertEquals(2, lraMetricService.getMetric(LRAMetricType.Nested, parentLRA, NestedParticipant.class));
+        assertEquals(0, lraMetricService.getMetric(LRAMetricType.AfterLRA, nestedLRA, NestedParticipant.class));
 
-        narayanaLRAClient.closeLRA(parentLRA);
+        lraClient.closeLRA(parentLRA);
         narayanaLRARecovery.waitForEndPhaseReplay(nestedLRA);
 
-        Assert.assertEquals("After LRA method for nested LRA enlist should have been called",
+        assertEquals("After LRA method for nested LRA enlist should have been called",
             1, lraMetricService.getMetric(LRAMetricType.AfterLRA, nestedLRA, NestedParticipant.class));
 
     }
+
+    @Test
+    public void testParentContext() {
+        // start a top level LRA
+        URI parent = lraClient.startLRA("parent");
+        // and nest another one under it
+        URI child = lraClient.startLRA("child");
+
+        Response response = null;
+
+        try {
+            response = client.target(UriBuilder.fromUri(baseURL.toExternalForm())
+                    .path(NestedParticipant.ROOT_PATH)
+                    .path(NestedParticipant.PATH))
+                    .request()
+//                .header(LRA.LRA_HTTP_CONTEXT_HEADER, parentLRA)
```

Comment:
Remove this please and add a comment explaining that the header is populated by the filter automatically?

Reply:
Sure. I see that the fix has broken one of the tests so I will remove it at the same time I resolve that failure. I'll also add a test for deeper transaction hierarchies.